lib/shellable.ts

```diff
@@ -268,8 +268,6 @@ export class Shellable extends Construct {
       build: this.platform.buildCommands(props.entrypoint),
     }).merge(props.buildSpec || BuildSpec.empty());
 
-    const environmentSecretsAsSecretNames = this.convertEnvironmentSecretArnsToSecretNames(props.environmentSecrets);
```

Comment:
Ever since https://github.com/aws/aws-cdk/pull/13706, we actually don't need this since our construct can accept an ARN as well. This was actually causing problems since there was conflicting logic, preventing a secret arn from being a token.

Namely, this was converting the values from an ARN to the secret name, which also ended up in the policy statement, but policy statements must be provided with an ARN.